llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -16613,11 +16613,56 @@ static SDValue performUADDVAddCombine(SDValue A, SelectionDAG &DAG) {
   return SDValue();
 }
 
+// We can convert a UADDV(add(zext(64-bit source), zext(64-bit source))) into
+// UADDLV(concat), where the concat represents the 64-bit zext sources.
+static SDValue performUADDVZextCombine(SDValue A, SelectionDAG &DAG) {
+  // Look for add(zext(64-bit source), zext(64-bit source)), returning
+  // UADDLV(concat(zext, zext)) if found.
+  if (A.getOpcode() != ISD::ADD)
```

Comment:
I don't think we need to check for ADD here because it was already checked in `performUADDVCombine`.

Reply:
Yeah, I took the check from performUADDVAddCombine too, but it is useless since we already check for that when we call the function, you're right. I'll remove it.